app/src/main/kotlin/compiler/common/dfa/state_dfa/TransparentDfa.kt

```diff
@@ -0,0 +1,10 @@
+package compiler.common.dfa.state_dfa
+
+import compiler.common.dfa.Dfa
+
+interface TransparentDfa<A, R> : Dfa<A, R> {
```

Comment:
Maybe `AbstractDfa` and `Dfa` instead of `Dfa` and `TransparentDfa`? (I suppose we'll use the second one more often.)

Reply:
This is better, but it still doesn't tell the difference between those two interfaces. We can rename it later if we came up with even better name.